Automattic/vip-cli
# Harmonia Pre Deploy Validation: Add `validate` command

## Description

Adds a new command - `validate` - that allows running Harmonia pre-deploy validation on any Node.js application. 

```
  Usage: vip validate.js [options] 
  
  Options:
    -a, --app           Specify the app
    -d, --debug         Activate debug output
    -e, --env           Specify the environment
    -h, --help          Output the help for the (sub)command
    -j, --json          Output the results as JSON (disabled by default)
    -n, --node-version  Select a specific target Node.JS version in semver format (MAJOR.MINOR.PATCH) or a MAJOR
    -o, --output        Output the results to a file
    -P, --path [value]  Path to the app to be tested (defaults to "/home/henrique/Repositories/vip-public-cli")
    -p, --port          Configure the port to use for the app (defaults to a random port between 3001 and 3999)
    -V, --verbose       Increase logging level to include app build and server boot up messages (disabled by default)
    -v, --version       Output the version number
    -w, --wait <n>      Configure the time to wait in ms for the app to boot up. Do not change unless you have issues (defaults to 3000)
  
  Examples:
    - Run checks and tests and validate your local environment against the production environment
    $ vip @mysite.production validate
```

~This PR is still work in progress. While the implementation works, there are still a few steps needed before it can be considered done:~

 - [x] Release Harmonia on  `@automattic/vip-go-preflight-checks` package
 - [x] Implement Parker query to automatically fetch all the required build data

## Steps to Test

1. Check out PR.
1. Run `npm install` to make sure the `@automattic/vip-go-preflight-checks` package is installed.
1. Run `npm run build`
1. Run `./dist/bin/vip-validate.js` or `npm link` and `vip @1234 validate`. Append a `--path` argument linking to the repository location ( `--path ~/Repositories/this-is-an-example`)
1. You should see Harmonia running the checks using the specified environment and code base.
